### ISSUE TITLE
Detect if the devTools are open in Chrome

### ISSUE THREAD

**Issue by mabasic** [wakatime/chrome-wakatime#15]
I have read the chrome API and there is no way to detect that directly. I will look for other solutions.

**Comment by teserak** [comment]
https://github.com/sindresorhus/devtools-detect

**Comment by mabasic** [comment]
@teserak I need to detect if the devTools are open from the background script of the extension in the current active  tab. Will that package work?

**Comment by teserak** [comment]
@mabasic Will work if you have access to window object. That package only check one condition: window.outerWidth - window.innerWidth > 160

**Comment by mabasic** [comment]
Ok, I'll give it a try

**Comment by alanhamlett** [comment]
Great find! Since you can access a page's JavaScript you can just run that script from the Chrome extension's content page script.

**Comment by mabasic** [comment]
@teserak I can't get this to work. It always returns `false`

**Comment by pritianka** [comment]
Hey @teserak - are you on the WakaTime slack channel?  I wanted to invite you to the builders call tomorrow at 11 AM PST where we will be discussing this chrome ext and also some other stuff people are working on.

**Comment by thecotne** [comment]
@teserak how `outerWidth`  and `innerWidth` is related to devtools ?
keep in mind that devtools can have separate window

**Comment by alanhamlett** [comment]
We've decided on using this to detect dev tools open/closed:
https://developer.chrome.com/extensions/devtools

**Comment by mabasic** [comment]
@alanhamlett  I have managed to get the tabId of the tab that has devTools open and store it in array, but I still have to find a way to detect when the devtools are closed on that specific tab. 

We cannot solve this like we discussed with messaging because it works on events and listeners and there is no way to wait for a response.

I will store tabId of the tabs that have devtools open in a variable and pass that to wakatime class.

I still have to find a way to detect when devtools for specific tab are closed so that I can remove that tabId from array.
